src/main/java/gregtech/api/recipes/Recipe.java

```diff
@@ -167,18 +167,13 @@ public boolean matches(boolean consumeIfSuccessful, List<ItemStack> inputs, List
 
         for (CountableIngredient ingredient : this.inputs) {
             int ingredientAmount = ingredient.getCount();
-            boolean isNotConsumed = false;
-            if (ingredientAmount == 0) {
-                ingredientAmount = 1;
-                isNotConsumed = true;
-            }
             for (int i = 0; i < inputs.size(); i++) {
                 ItemStack inputStack = inputs.get(i);
                 if (inputStack.isEmpty() || !ingredient.getIngredient().apply(inputStack))
                     continue;
                 int itemAmountToConsume = Math.min(itemAmountInSlot[i], ingredientAmount);
                 ingredientAmount -= itemAmountToConsume;
```

Comment:
Should this also be behind a check for if the ingredient is not consumable? As do we want to substract from the ingredient in the input if the itemStack should not be consumed at all?

Reply:
the check is still in place, just below. 
                if (!ingredient.isNonConsumable()) itemAmountInSlot[i] -= itemAmountToConsume;

Reply:
yeah, that is subtracting from the amount of items in a certain slot. I was talking about subtracting from `ingredientAmount`, which is the amount in the recipe input